ReadMe.md

```diff
@@ -1,3 +1,116 @@
 # MAGE SDK for JavaScript (ES5)
 
-This is a work-in-progress.
+This is the official MAGE SDK for the web. It is to be used in conjunction with MAGE server
+applications. You can install [MAGE](https://www.npmjs.com/package/mage) from NPM.
+
+This SDK is completely written in ES5 to be compatible with the widest range of browsers.
+It uses the CommonJS style (`require()` calls and `module.exports`), and can thus be used with
+build systems like [Webpack](https://webpack.github.io/), which is the one we use and test on.
+
+## Compatibility
+
+This version of `mage-sdk-js` works with **MAGE >=1.1.0**.
+
+## Installation
+
+In the examples we use, we will assume your web code will live in the same project folder as
+your server application.
+
+```shell
+npm install --save mage-sdk-js
+```
+
+If you also want to install some MAGE module clients, you can install them via NPM as well:
+
+```shell
+npm install --save mage-sdk-js.session
+npm install --save mage-sdk-js.logger
+npm install --save mage-sdk-js.time
+npm install --save mage-sdk-js.archivist
+```
+
+## Setup
+
+If you have bootstrapped a MAGE server application, you need to make sure that the MAGE
+configuration ends up in the build system. Among other things, because the client needs to be
+aware of all the user commands the server exposes.
+
+### Example
+
+The example below assumes your project's `index.js` file setting up MAGE and exporting the
+`mage` object, but you can easily see how you could customize the setup to your liking.
+
+> lib/index.js
+
+```javascript
+'use strict';
+
+const mage = module.exports = require('mage');
+
+mage.useModules([
+	'session',
+	'auth',
+	'logger',
+	'time',
+	'archivist'
+]);
+
+mage.useApplicationModules();
+```
+
+#### Webpack
+
+A typical Webpack configuration exposing the MAGE configuration to the client may look like this:
+
+> webpack.config.js
+
+```javascript
+'use strict';
+
+const webpack = require('webpack');
+const mage = require('./lib');
+
+const appName = 'game';
+
+module.exports = {
+	entry: {
+		index: './www/src/index.js'
+	},
+	output: {
+		path: './www/dist',
+		filename: '[name].js'
+	},
+	module: {
+		loaders: [
+			{ test: /\.css$/, loader: 'css-loader' },
+			{ test: /\.html?$/, loader: 'html-loader' },
+			{ test: /\.js$/, loader: 'strict-loader' }
+		]
+	},
+	plugins: [
+		new webpack.DefinePlugin({
+			MAGE_CONFIG: JSON.stringify(mage.getClientConfig(appName))
+		})
+	]
+};
+```
+
+Your browser code needs to expose the MAGE configuration to `mage-sdk-js`, which it does by assigning it to
+`window.mageConfig`. Your code may look somewhat like this:
+
+> www/src/index.js
+
+```javascript
+window.mageConfig = MAGE_CONFIG;
+
+var mage = require('mage-sdk-js');
+
+mage.addModule('session', require('mage-sdk-js.session'));
+mage.addModule('logger', require('mage-sdk-js.logger'));
+mage.addModule('time', require('mage-sdk-js.time'));
+mage.addModule('archivist', require('mage-sdk-js.archivist'));
+
+mage.setup(function (error) {
```

Comment:
Arguably, should be in a docs, but if we show how setup works we might as well demo a single call just to give a taste of things to come in the documentation.